src/main/java/com/empowerops/linqalike/DefaultedQueryable.java

```diff
@@ -266,11 +266,11 @@ TElement withMin(Func1<? super TElement, TCompared> valueSelector){
 
 
     /** {@inheritDoc} */ @Override default public
-	Queryable<Tuple<TElement, TElement>> pairwise(){
+	BiQueryable<TElement, TElement> pairwise(){
         return Linq.pairwise(this);
     }
     /** {@inheritDoc} */ @Override default public
-	Queryable<Tuple<TElement, TElement>> pairwise(Func<? extends TElement> defaultFactory){
+	BiQueryable<TElement, TElement> pairwise(Func<? extends TElement> defaultFactory){
         return Linq.pairwise(this, defaultFactory);
```

Comment:
ahh yes, and these methods, I've got 2 use cases on OASIS and iirc both of them don't particularly deal well with the end cases. Not sure what to do here.